packages/opentelemetry-cloud-monitoring-exporter/src/monitoring.ts

```diff
@@ -93,7 +93,13 @@ export class MetricExporter implements IMetricExporter {
     cb: (result: ExportResult) => void
   ): Promise<void> {
     if (this._projectId instanceof Promise) {
-      this._projectId = await this._projectId;
+      try {
```

Comment:
Nice! I think the await on L114 could also throw. Maybe we could do something like this to generalize:

```ts
  async _exportAsync(
    metrics: MetricRecord[],
  ): Promise<ExportResult> {
    // this comes from renaming and updating params for the existing export() function
    // and instead return the ExportResult
  }
  
  export(
    metrics: MetricRecord[],
    cb: (result: ExportResult) => void
  ): void {
    _exportAsync(metrics)
      .then(cb)
      .except(err => {
        diag.error(err.message);
        cb(/* map err to ExportResult */);
      });
  }
```

Reply:
Oh wow yeah great catch. I totally missed the await on L114. Great idea on wrapping the async calls in their own function. I'll get on that asap.

Reply:
@aabmass apologies for the delay here. Lmk how this looks. I added some comments for clarity. I'm happy to add some more tests too if necessary.

Reply:
No worries. I think you can remove this new try block since it will be caught in the `.catch()` above.